services/repository/files/patch.go

```diff
@@ -164,9 +164,9 @@ func ApplyDiffPatch(ctx context.Context, repo *repo_model.Repository, doer *user
 	// Now commit the tree
 	var commitHash string
 	if opts.Dates != nil {
-		commitHash, err = t.CommitTreeWithDate(author, committer, treeHash, message, opts.Signoff, opts.Dates.Author, opts.Dates.Committer)
+		commitHash, err = t.CommitTreeWithDate("HEAD", author, committer, treeHash, message, opts.Signoff, opts.Dates.Author, opts.Dates.Committer)
```

Comment:
```suggestion
		commitHash, err = t.CommitTreeWithDate(repo.DefaultBranch, author, committer, treeHash, message, opts.Signoff, opts.Dates.Author, opts.Dates.Committer)
```
?

Reply:
The "current" behavior is using "HEAD", https://github.com/go-gitea/gitea/pull/19224/files#diff-b0213d5f52f1ddd2d9701c3f073f147a0a18f76d8975605dc5037a377a25f24aL238 

It uploads to a temporary repository, given it clones to a new repo and the existing repo can have any kind of default branch name the "HEAD" seems to be most correct here(given we just clone, so the HEAD should be set correct).

Reply:
noticed while reading code ...